corehq/apps/domain/forms.py

```diff
@@ -472,8 +480,26 @@ def clean_default_timezone(self):
         timezone_field.run_validators(data)
         return smart_str(data)
 
+    def clean_alias(self):
+        data = self.cleaned_data['alias'].strip().lower()
+        if data == "":
+            return data
+
+        if not re.match("^%s$" % new_domain_re, data):
+            raise forms.ValidationError('Only lowercase letters and numbers allowed. ' +
+                'Single hyphens may be used to separate words.')
+
+        data_alternate = data.replace('-', '.')
+        conflict = Domain.get_by_name(data) or Domain.get_by_name(data_alternate)
```

Comment:
data & data_alternate? I don't actually know, stole that logic from https://github.com/dimagi/commcare-hq/blame/master/corehq/apps/registration/forms.py#L145 Seems to have been introduced with organizations, but I'm not sure why.